### PR TITLE
Add some new Columns & Tables

### PR DESCRIPTION
**Pull request by boehsermoe**
Add a config table e.g to upload a default placeholder image for products.
And add `enabled`, `visibility` and `code` columns to some tables.

### CODE REVIEW

File: src/admin/migrations/m170515_115236_basetables.php [hunk 1]
```diff
@@ -22,29 +22,35 @@ public function safeUp()
...
         $this->addPrimaryKey('estore_product_set_ref_pk', '{{%estore_product_set_ref}}', ['product_id', 'set_id']);
-        
+        $this->addForeignKey('estore_product_set_ref_product_id_fk', '{{%estore_product_set_ref}}', ['product_id'], '{{%estore_product}}', ['id'], 'CASCADE', 'CASCADE');
+        $this->addForeignKey('estore_product_set_ref_set_id_fk', '{{%estore_product_set_ref}}', ['set_id'], '{{%estore_set}}', ['id'], 'CASCADE', 'CASCADE');
+
         $this->createTable('{{%estore_set_attribute}}', [
             'id' => $this->primaryKey(),
             'type' => $this->integer(), // 1 = integer, 2 = boolean, 3 = string
             'input' => $this->string()->notNull(), // zaa-text, zaa-password
+            'code' => $this->string()->notNull(), // size, color, material_type, width, height
```
Comment (nadar) [inline src/admin/migrations/m170515_115236_basetables.php]: Its a slug? should be unique? For url generation? or maybe its "identifier"?

**Review by nadar — COMMENTED** [review]
In general this looks good to me!

File: src/admin/migrations/m170515_115236_basetables.php (hunk 1 above)
Comment (boehsermoe) [reply src/admin/migrations/m170515_115236_basetables.php]: yes right should be also unique.

**Comment by nadar** [comment]
@boehsermoe I just did a small review again on those prs #6 #8 #9  - this is by far to complex to merge - i think we should have smaller pull requests which also explains the introduction of the given changes. Why change from article to product price? why remove article in menu? Taking care of the new fields - why `code` instead of alias or identifier (or whatever).  Why introduce new Config, maybe use luya\admin\Config? etc. The other two PRs are even more complexity.

I have added two things which i copied from your code: https://github.com/luyadev/luya-module-estore/commit/d2c932b198b8ac470701f88dca5d57d479b6db22

But i will close those PRs - i am really sorry about! PRs are highly welcome, but we should have smaller change request.

**Comment by boehsermoe** [comment]
Ok sad to hear but I understand it. Maybe I find time to create smaller PRs of it. 
Thank you for your effort.

**Comment by nadar** [comment]
@boehsermoe yes, it was also sad for me to make this decision as i can see all the time and effort you spent into to changes. I am sorry. But i hope you can understand why i made this decision. Mabye we can create an issue an write down all the improvements you made (point by point) and generate a PR for every point (based on the existing code).

**Pull request closed without merging**